placementApp/views.py

```diff
@@ -1,10 +1,43 @@
-from rest_framework import viewsets, permissions, mixins
 import xlwt
 import datetime
 from django.shortcuts import HttpResponse
-
 from .models import Student, Position, Company
-from .serializers import StudentSerializer, PositionSerializer
+from .serializers import *
+from django.contrib.auth import get_user_model
+from rest_framework import viewsets, permissions, status,mixins,generics
+from django.contrib.auth.hashers import make_password
+from rest_framework.response import Response
+
+
+class StudentSignUpView(generics.CreateAPIView):
+    permission_classes = (permissions.AllowAny,)
+    queryset = Student.objects.all()
+    serializer_class = StudentSignupSerializer
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.validated_data["role"] = "STUDENT"
```

Comment:
One error that I'm getting while student signup is TypeError because the serializer object has a password2 field while the actual student model doesn't one way to handle it is by simply doing serializer.validated_data.pop("password2") at this point after checking that password and password2 matches or not